pkgs/tools/security/pinentry/default.nix

```diff
@@ -22,6 +22,7 @@ let
       lib.optionalString (flag != null)
         (lib.enableFeature (lib.elem f enabledFlavors) ("pinentry-" + flag));
 
+  # keep in sync with programs.gnupg.agent.pinentryFlavor
```

Comment:
How about putting in this package an `assert lib.attrNames flavorInfo == import ./flavor-list.nix;`, so that you can import that file from NixOS?

Reply:
What about using a `passthru.flavors = lib.attrNames flavorInfo`, or is using a package's passthru attributes in NixOS options disallowed?

Reply:
I've implemented the second option as this added the least amount of code.

Reply:
Normally, any access to `pkgs` would be disallowed but `gnupg.nix` is opting out of sandboxing because the module was already using `passthru`:

https://github.com/NixOS/nixpkgs/blob/0b3fbec39d1426331e57c0e289d20a6911218c12/nixos/modules/programs/gnupg.nix#L238-L239